src/infrastructure/content/NetworkMessageHandler.java

```diff
@@ -0,0 +1,151 @@
+	 */
+	private String userimage;
+	
+	/**
+	* This variable will store a string which will contains log message that to be passed to log method of logger
+	*/
+	private String logMessage;
+	
+	/**
+	 * Getting handler of the UI module to send the data
+	 */
+	private IContentNotificationHandler handler;
+	 
+	/**
+	 * @param String message - JSON string message (Meta fields are newUser or message or userExit)
+	 *
+	 * if meta field is newUser, then remaining field of parameter message would only be username,  
+	 * to send this message to UI removing metafield, adding image and then calling @onNewUserJoined() method
+	 * if meta field is message, then remaining fields of parameter message would be username, message and time, 
+	 * to send this message to UI removing metafield, adding image and then calling @onMessageReceived() method
+	 * if meta field is exitUser, then remaining field of parameter message would only be username, 
+	 * to send this message to UI removing metafield, adding image and then calling @onNewUserJoined() method
+	 * Networking module will be calling this method with json string message to send data to the content module
+	 *
+	 * creating error log message if message in not converting into json object,
+	 * or if not able extract field from object, or if not able to extract handler from hashmap (handlerMap)
+	 * if message will be successfully sent to UI, then creating success log message
+	 */
+	@override
+	public void onMessageReceived(String message) {
+		try {
+			JSONObject jsonObject = new JSONObject(message);
+		} 
+		catch(Exception e) {
+			logMessage = "converting JSON string to JSON object process failed";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+		
+		try {
+			metafield = jsonObject.getString("meta");
+		}
+		catch(Exception e) {
+			logMessage = "failed to get metafield data of message parameter";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+		
+		try {
+			handler = handlerMap.get("UI");
+		}
+		catch(Exception e) {
+			logMessage = "failed to get IContentNotification handler from handlerMap";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+		
+		try {
+			username = jsonObject.getString("username");
+		}
+		catch(Exception e) {
+			logMessage = "failed to get username field data of message parameter";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+		
+		try {
+			userimage = imageMap.get(username);
+		}
+		catch(Exception e) {
+			logMessage = "failed to get image from imageMap";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+		
+		jsonObject.remove("meta");
+		jsonObject.put("image", userimage);
+		
+		if(metafield.equals("newUser")) {
+			handler.onNewUserJoined(jsonObject.toString());
+			logMessage = "onNewUserJoined() called";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
```

Comment:
Logic for updating imageMap on ContentCommunicator is not present. Badal is sending an array of JSON objects. Common logic for all meta cases is not going to work. You need to create a new JSON object for newUser case